src/rpc/masternode.cpp

```diff
@@ -324,6 +324,148 @@ UniValue masternode_winners(const JSONRPCRequest& request)
 
     return obj;
 }
+void masternode_payments_help()
+{
+    throw std::runtime_error(
+            "masternode payments ( \"blockhash\" count )\n"
+            "\nReturns an array of deterministic masternodes and their payments for the specified block\n"
+            "\nArguments:\n"
+            "1. \"blockhash\"                       (string, optional, default=tip) The hash of the starting block\n"
+            "2. count                             (numeric, optional, default=1) The number of blocks to return.\n"
+            "                                     Will return <count> previous blocks if <count> is negative.\n"
+            "                                     Both 1 and -1 correspond to the chain tip.\n"
+            "\nResult:\n"
+            "  [                                  (array) Blocks\n"
+            "    {\n"
+            "       \"height\" : n,                 (numeric) The height of the block\n"
+            "       \"blockhash\" : \"hash\",         (string) The hash of the block\n"
+            "       \"amount\": n                   (numeric) Amount received in this block by all masternodes\n"
+            "       \"masternodes\": [              (array) Masternodes that received payments in this block\n"
+            "          {\n"
+            "             \"proTxHash\": \"xxxx\",    (string) The hash of the corresponding ProRegTx\n"
+            "             \"amount\": n             (numeric) Amount received by this masternode\n"
+            "             \"payees\": [             (array) Payees who received a share of this payment\n"
+            "                {\n"
+            "                  \"address\" : \"xxx\", (string) Payee address\n"
+            "                  \"script\" : \"xxx\",  (string) Payee scriptPubKey\n"
+            "                  \"amount\": n        (numeric) Amount received by this payee\n"
+            "                },...\n"
+            "             ]\n"
+            "          },...\n"
+            "       ]\n"
+            "    },...\n"
+            "  ]\n"
+        );
+}
+
+UniValue masternode_payments(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() > 3) {
+        masternode_payments_help();
+    }
+
+    CBlockIndex* pindex{nullptr};
+
+    if (request.params[1].isNull()) {
+        LOCK(cs_main);
+        pindex = chainActive.Tip();
+    } else {
+        LOCK(cs_main);
+        uint256 blockHash = ParseHashV(request.params[1], "blockHash");
```

Comment:
```suggestion
        uint256 blockHash = ParseHashV(request.params[1], "blockhash");
```

nit nit, to match the parameter description of the help string?